packages/auth-kit/src/packs/web3auth/Web3AuthAdapter.ts

```diff
@@ -56,22 +56,40 @@ export class Web3AuthAdapter implements SafeAuthAdapter<Web3AuthAdapter> {
    * Connect to the Web3Auth service provider
    * @returns
    */
-  async signIn(): Promise<void> {
-    if (!this.web3authInstance) return
+  async signIn() {
+    if (!this.web3authInstance) {
+      throw new Error('Web3AuthAdapter is not initialized')
+    }
 
     this.provider = await this.web3authInstance.connect()
   }
 
   /**
    * Disconnect from the Web3Auth service provider
    */
-  async signOut(): Promise<void> {
-    if (!this.web3authInstance) return
+  async signOut() {
+    if (!this.web3authInstance) {
+      throw new Error('Web3AuthAdapter is not initialized')
+    }
 
     this.provider = null
     await this.web3authInstance.logout()
   }
 
+  /**
+   * Get authenticated user information
+   * @returns The user info
+   */
+  async getUserInfo(): Promise<Partial<UserInfo>> {
+    if (!this.web3authInstance) {
+      throw new Error('Web3AuthAdapter is not initialized')
+    }
+
+    const userInfo = await this.web3authInstance?.getUserInfo()
```

Comment:
This should not be needed because of the previous check
```suggestion
    const userInfo = await this.web3authInstance.getUserInfo()
```